### PR TITLE
add mike

### PR DESCRIPTION
**Pull request by hadim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by hadim** [comment]
@conda-forge/help-python please review

File: recipes/mike/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - pip check
+    - mike --help
+  requires:
+    - pip
+
+about:
+  home: https://github.com/jimporter/mike
+  summary: Manage multiple versions of your MkDocs-powered documentation
+  license: BSD-3-Clause
+  license_file: LICENSE
```
Comment (synapticarbors) [inline recipes/mike/meta.yaml:47]: Please add a comment:

```yaml
# License file manually packaged. See https://github.com/jimporter/mike/pull/23
license_file: LICENSE
...
```

I've opened a PR to fix the need to manually package the license file (https://github.com/jimporter/mike/pull/23). The comment as a reminder to check the status of this for future updates in the feedstock. Once it's in the source distribution, the local copy can be removed.

File: recipes/mike/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+build:
+  number: 0
+  noarch: python
+  entry_points:
+    - mike = mike.driver:main
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - flake8
```
Comment (synapticarbors) [inline recipes/mike/meta.yaml:22]: Are `flake8` and `pypandoc` needed under `host`? I see that `pypandoc` is optionally needed to get the long description, but neither will cause `setup.py` to fail.

File: recipes/mike/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - pip
+    - pypandoc
+    - python
```
Comment (synapticarbors) [inline recipes/mike/meta.yaml:25]: Make `python` -> `python >=3.5` under `host` and `run` since the project doesn't have python2 support.

**Comment by hadim** [comment]
Thank you @synapticarbors for the review. CI is green.

Do you mind having a look at https://github.com/conda-forge/staged-recipes/pull/13002 too?

**Pull request merged**